thingsboard_gateway/tb_utility/tb_gateway_remote_configurator.py

```diff
@@ -136,8 +142,14 @@ def __prepare_connectors_configuration(self, input_connector_config):
                     if input_connector['name'] == connector['name']:
                         if not self.__gateway.connectors_configs.get(connector['type']):
                             self.__gateway.connectors_configs[connector['type']] = []
+                        config_file_path = self.__gateway._config_dir + connector['configuration']
+                        # Create the configuration json file if not exists
+                        open(config_file_path, 'w')
                         self.__gateway.connectors_configs[connector['type']].append(
-                            {"name": connector["name"], "config": {connector['configuration']: input_connector["config"]}})
+                            {"name": connector["name"], 
+                            "config": {connector['configuration']: input_connector["config"]},
+                            "config_updated": stat(config_file_path),
```

Comment:
In tb_gateway_service.py line 455-460

>                     self.connectors_configs[connector['type']].append({"name": connector['name'],
>                                                                        "config": {connector['configuration']: connector_conf} if connector[
>                                                                                                                                      'type'] != 'grpc' else connector_conf,
>                                                                        "config_updated": stat(config_file_path),
>                                                                        "config_file_path": config_file_path,
>                                                                        "grpc_key": connector_persistent_key})

and if `config_updated` is not defined, it will run into a KeyError in `check_connector_configuration_updates`

> def check_connector_configuration_updates(self):
>         ...
>                 if stat(connector_config["**config_file_path**"]) != connector_config["**config_updated**"]:
>                     configuration_changed = True
>                     break
>            ...

Reply:
Agreed, didn't notice this.